### ISSUE TITLE
ECS Provider -- autoDiscoverClusters option behavior does not match the documentation.

### ISSUE THREAD

**Issue by johnpekcan** [traefik/traefik#9385]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

I configured Traefik ECS Provider using ENV vars as documented in [here](https://doc.traefik.io/traefik/reference/static-configuration/env/).
I referenced the documentation regarding auto discovery provided [here](https://doc.traefik.io/traefik/providers/ecs/#autodiscoverclusters):
> If set to true service discovery is disabled on configured clusters, but enabled for all other clusters.
   If set to false service discovery is enabled on configured clusters only.

Based on this I expected this configuration:
```
                "TRAEFIK_PROVIDERS_ECS_AUTODISCOVERCLUSTERS": "true",
                "TRAEFIK_PROVIDERS_ECS_CLUSTERS": "test1",
```
To read as: Go ahead and auto discover clusters, but exclude `test1` 

### What did you see instead?

Instead, I saw the config I provided be read back to me in `TRACE`:

```
- TRAEFIK_PROVIDERS_ECS_AUTODISCOVERCLUSTERS=true
- TRAEFIK_PROVIDERS_ECS_CLUSTERS=test1

time="2022-09-27T17:33:03Z" level=info msg="Starting provider *ecs.Provider"
time="2022-09-27T17:33:03Z" level=debug msg="*ecs.Provider provider configuration: {\"exposedByDefault\":true,\"refreshSeconds\":15,\"defaultRule\":\"Host(`{{ normalize .Name }}`)\",\"clusters\":[\"test1\"],\"autoDiscoverClusters\":true,\"region\":\"us-west-2\"}"
time="2022-09-27T17:33:04Z" level=debug msg="ECS Clusters: [arn:aws:ecs:us-west-2:REDACTED:cluster/default arn:aws:ecs:us-west-2:REDACTED:cluster/test1 arn:aws:ecs:us-west-2:REDACTED:cluster/test2 arn:aws:ecs:us-west-2:REDACTED:cluster/test3 arn:aws:ecs:us-west-2:REDACTED:cluster/test4]" providerName=ecs
```

^ I expected the above config to just return `test1` as the ECS Clusters list. Based on my understanding of the documentation.

```
- TRAEFIK_PROVIDERS_ECS_AUTODISCOVERCLUSTERS=false
- TRAEFIK_PROVIDERS_ECS_CLUSTERS=test1

time="2022-09-27T17:33:41Z" level=info msg="Starting provider *ecs.Provider"
time="2022-09-27T17:33:41Z" level=debug msg="*ecs.Provider provider configuration: {\"exposedByDefault\":true,\"refreshSeconds\":15,\"defaultRule\":\"Host(`{{ normalize .Name }}`)\",\"clusters\":[\"test1\"],\"region\":\"us-west-2\"}"
time="2022-09-27T17:33:41Z" level=debug msg="ECS Clusters: [test1]" providerName=ecs
```

^ This does appear to be working as intended. Do not go auto discovering, simply use the cluster name I give to you.

### What version of Traefik are you using?

Released version: v2.8.7 as well as the main branch in the repo.

```
➜  traefik git:(master) ✗ ./traefik version  
Version:      dev
Codename:     cheddar
Go version:   go1.19
Built:        I don't remember exactly
OS/Arch:      darwin/arm64

top of my gitlog entry:
commit 033fccccc7552de4c41aa906c86bf27d072aebe6 (HEAD -> master, origin/master, origin/HEAD)
Author: jjacque <jeremy.jacque@algolia.com>
Date:   Tue Sep 20 16:54:08 2022 +0200

    Support gRPC healthcheck
```

### What is your environment & configuration?

```
➜  ✗ screenfetch

                 -/+:.          @johnpekcan
                :++++.          OS: 64bit Mac OS X 12.5.1 21G83
               /+++/.           Kernel: arm64 Darwin 21.6.0
       .:-::- .+/:-``.::-       Uptime: 5d 16h 43m
    .:/++++++/::::/++++++/:`    Packages: 85
  .:///////////////////////:`   Shell: bash 5.1.16
  ////////////////////////`     Resolution: 3440x1440
 -+++++++++++++++++++++++`      DE: Aqua
 /++++++++++++++++++++++/       WM: Quartz Compositor
 /sssssssssssssssssssssss.      WM Theme: Blue (Dark)
 :ssssssssssssssssssssssss-     Disk: 23G / 995G (3%)
  osssssssssssssssssssssssso/`  CPU: Apple M1 Pro
  `syyyyyyyyyyyyyyyyyyyyyyyy+`  GPU: Apple M1 Pro
   `ossssssssssssssssssssss/    RAM: 3508MiB / 32768MiB
     :ooooooooooooooooooo+.
      `:+oo+/:-..-:/+o+/-
```

VS Code launch.json:
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Launch Package",
            "type": "go",
            "request": "launch",
            "mode": "auto",
            "program": "cmd/traefik",
            "env": {
                "AWS_ACCESS_KEY_ID": "REDACTED",
                "AWS_SECRET_ACCESS_KEY": "REDACTED",
                "AWS_SESSION_TOKEN": "REDACTED",
                "AWS_REGION": "us-west-2",
                "AWS_PROFILE": "my_profile",
                "TRAEFIK_PROVIDERS_ECS_REGION": "us-west-2",
                "TRAEFIK_PROVIDERS_ECS": "true",
                "TRAEFIK_PROVIDERS_ECS_AUTODISCOVERCLUSTERS": "true",
                "TRAEFIK_PROVIDERS_ECS_CLUSTERS": "test1",
                "TRAEFIK_API": "true",
                "TRAEFIK_API_INSECURE": "true",
                "TRAEFIK_LOG_LEVEL": "TRACE",
              }
        }
    ]
}
```

### If applicable, please paste the log output in DEBUG level

Log lines provided above.

**Comment by johnpekcan** [comment]
Ah, that makes sense! Thanks @kevinpollet  I'll get a PR going for that shortly.
I did have a PR that enables this behavior if we were interested 👀 #9386

**Comment by kevinpollet** [comment]
Closed by #9392
